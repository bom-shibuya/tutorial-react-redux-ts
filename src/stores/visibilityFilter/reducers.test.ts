import * as snapshotDiff from 'snapshot-diff'
import * as reducers from '.'

describe('VisibilityFilter reducers test', () => {
  const {
    initialState,
    visibilityFilter,
    VisibilityFilters,
    setVisibilityFilter
  } = reducers

  test('should return initial state', () => {
    expect(visibilityFilter(initialState, {} as any)).toBe(
      VisibilityFilters.SHOW_ALL
    )
  })

  test('SHOW_ALL', () => {
    expect(
      snapshotDiff(
        initialState,
        visibilityFilter(
          initialState,
          setVisibilityFilter(VisibilityFilters.SHOW_ALL)
        )
      )
    ).toMatchSnapshot()
  })

  test('SHOW_ACTIVE', () => {
    expect(
      snapshotDiff(
        initialState,
        visibilityFilter(
          initialState,
          setVisibilityFilter(VisibilityFilters.SHOW_ACTIVE)
        )
      )
    ).toMatchSnapshot()
  })

  test('SHOW_COMPLETED', () => {
    expect(
      snapshotDiff(
        initialState,
        visibilityFilter(
          initialState,
          setVisibilityFilter(VisibilityFilters.SHOW_COMPLETED)
        )
      )
    ).toMatchSnapshot()
  })
})
