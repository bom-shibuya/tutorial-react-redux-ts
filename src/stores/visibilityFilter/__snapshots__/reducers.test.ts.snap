// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisibilityFilter reducers test SHOW_ACTIVE 1`] = `
"Snapshot Diff:
- First value
+ Second value

- SHOW_ALL
+ SHOW_ACTIVE"
`;

exports[`VisibilityFilter reducers test SHOW_ALL 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`VisibilityFilter reducers test SHOW_COMPLETED 1`] = `
"Snapshot Diff:
- First value
+ Second value

- SHOW_ALL
+ SHOW_COMPLETED"
`;
